src/sdl2/render.rs

```diff
@@ -380,13 +519,18 @@ impl Error for TextureValueError {
 }
 
 /// Texture-creating methods for the renderer
-impl<'a> Renderer<'a> {
+impl<T> TextureCreator<T> {
```

Comment:
The textures now have a lifetime to prevent them from living past whoever created them. If we let the user create textures using the `Canvas` directly, then the lifetime borrow will be on the `Canvas`. This means that after any `Texture` has been created, no `&mut` actions can be done on the canvas, including modifying the window, or even drawing the `Texture` itself. 

What is a case where you think we would want to create a `Texture` from the `Canvas` directly? 
Also, how do you envision this being done without a direct copy-paste of the code of `TextureCreator` onto the `Canvas`?